Classes/Event/EventPublisher.php

```diff
@@ -94,8 +94,12 @@ public function publishMany(string $streamName, array $events, int $expectedVers
     {
         $convertedEvents = new WritableEvents();
         foreach ($events as $event) {
-            $type = $this->eventTypeResolver->getEventType($event);
             $metadata = [];
+            if ($event instanceof EventWithMetadata) {
```

Comment:
I'd suggest to check against an interface here, so you can implement that in your domain event implementation

Reply:
👍

Reply:
The reason why I didn't introduce the interface yet was that I wasn't sure if we really, really want to keep this concept as is in the future.

Anyway, I wonder why you'd implement this interface in your domain event directly? `getEvent()` would then return `$this`?

Reply:
Good point, I didn't think of that - the interface would need to be checked in the conversion back from the ES, so that the metadata could be set from the `RawEvent`..